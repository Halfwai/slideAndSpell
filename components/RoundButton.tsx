import { TouchableOpacity, StyleSheet } from "react-native";
import MaterialIcon from "react-native-vector-icons/MaterialCommunityIcons";
import EntypoIcon from "react-native-vector-icons/Entypo";
import { COLOURS } from "@/constants/colours";
import * as Haptics from 'expo-haptics';
import { useContext } from "react";
import { UserContext } from "@/utils/context";


interface RoundButtonProps {
  icon: string;
  onPress: () => void;
  iconType?: "material" | "entypo";
  colour?: string;
}

export default function RoundButton(props: RoundButtonProps) {
    const userContext = useContext(UserContext);
    return (
        <TouchableOpacity 
            style={[styles.button, props.colour && {backgroundColor: props.colour}]} 
            onPress={async () => {
                if (userContext && userContext.vibrate) {
                    await Haptics.impactAsync(Haptics.ImpactFeedbackStyle.Light);
                }
                props.onPress()
            }}
        >
        {props.iconType === "entypo" ? (
            <EntypoIcon name={props.icon} size={30} color={"black"} />
        ) : (
            <MaterialIcon name={props.icon} size={30} color={"black"} />
        )}
        </TouchableOpacity>
    );
}

const styles = StyleSheet.create({
  button: {
    backgroundColor: "white",
    borderColor: COLOURS.green,
    borderWidth: 2,
    borderRadius: 50,
    padding: 15,
  },
});
