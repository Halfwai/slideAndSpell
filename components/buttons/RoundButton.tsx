import { TouchableOpacity, StyleSheet } from "react-native";
import { MaterialCommunityIcons, Entypo } from "@expo/vector-icons";
import { COLOURS } from "@/constants/colours";
import * as Haptics from 'expo-haptics';
import React, { useContext } from "react";
import { UserContext } from "@/utils/context";


interface RoundButtonProps {
  icon: string;
  onPress: () => void;
  iconType?: "material" | "entypo";
  colour?: string;
}

export default function RoundButton(props: RoundButtonProps) {
    const userContext = useContext(UserContext);
    return (
        <TouchableOpacity 
            style={[styles.button, props.colour && {backgroundColor: props.colour}]} 
            onPress={async () => {
                if (userContext && userContext.vibrate) {
                    await Haptics.impactAsync(Haptics.ImpactFeedbackStyle.Light);
                }
                props.onPress()
            }}
            testID="roundButton"
        >
        {props.iconType === "entypo" ? (
            <Entypo name={props.icon as any} size={30} color={"black"} />
        ) : (
            <MaterialCommunityIcons name={props.icon as any} size={30} color={"black"} />
        )}
        </TouchableOpacity>
    );
}

const styles = StyleSheet.create({
  button: {
    backgroundColor: "white",
    borderColor: COLOURS.green,
    borderWidth: 2,
    borderRadius: 50,
    padding: 15,
  },
});
