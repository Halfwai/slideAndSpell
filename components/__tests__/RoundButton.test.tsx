import { render, fireEvent } from '@testing-library/react-native';

import RoundButton from '@/components/buttons/RoundButton';

afterEach(() => {
    jest.clearAllMocks();
    jest.useRealTimers();
});

it(`RoundButton Component renders correctly`, () => {
    const { toJSON } = render(
        <RoundButton 
            icon={"lightbulb-on-outline"}
            onPress={() => {}}
            iconType={"material"}
            colour={"#000"}
        />
    );
    expect(toJSON()).toMatchSnapshot();
});

it(`RoundButton press calls function`, () => {
    const testFunction = jest.fn()
    const { getByTestId } = render(
        <RoundButton 
        icon={"lightbulb-on-outline"}
        onPress={testFunction}
        iconType={"material"}
        colour={"#000"}
    />
    );
    const button = getByTestId('roundButton');
    fireEvent.press(button);
    expect(testFunction).toHaveBeenCalled();
});