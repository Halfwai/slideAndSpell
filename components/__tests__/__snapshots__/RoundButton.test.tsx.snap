// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundButton Component renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#000",
      "borderColor": "#8BF0BC",
      "borderRadius": 50,
      "borderWidth": 2,
      "opacity": 1,
      "padding": 15,
    }
  }
  testID="roundButton"
>
  MaterialCommunityIconsMock
</View>
`;
